confirmation_rule.py

```diff
@@ -125,7 +125,10 @@ def find_latest_confirmed_descendant(store, latest_confirmed_root) -> Root:
         )
         
         if (block_epoch > compute_epoch_at_slot(latest_confirmed_slot)
-            and not will_checkpoint_be_justified(store, checkpoint)):
+            and 
+                (not will_checkpoint_be_justified(store, checkpoint) or
+                store.block_states[get_head(store)].current_justified_checkpoint != checkpoint)
```

Comment:
this will always be `False`, I guess you meant:
```suggestion
                store.unrealized_justifications[get_head(store)] != checkpoint)
```